code/modules/mining/minebot.dm

```diff
@@ -75,8 +75,9 @@
 			if(maxHealth == health)
 				user << "<span class='info'>[src] is at full integrity.</span>"
 			else
-				adjustBruteLoss(-10)
-				user << "<span class='info'>You repair some of the armor on [src].</span>"
+				if(W.remove_fuel(1, user))
+					adjustBruteLoss(-10)
+					user << "<span class='info'>You repair some of the armor on [src].</span>"
```

Comment:
1 unit of fuel is removed in /obj/item/weapon/weldingtool/afterattack(), return 1 here to avoid doubling the cost.